app/Main.hs

```diff
@@ -189,14 +188,21 @@ zmqInterruptibleSource m s = do
     liftIO $ putStrLn "Done!"
 
 --------------------------------------------------------------------------------
+normalisationConduit options config =
+    let fs = fields config
+    in if oJsonInput options
+        then CB.lines $= C.map (normaliseJsonInput fs)
+        else DCT.decode DCT.utf8 $= DCT.lines $= C.map (normaliseText fs)
```

Comment:
this `if-then-else` is an exact duplicate of the one used above (twice), so add a function for it?

Reply:
There's an inferred type difference, so that will not work :)

Reply:
haha, that's aweful (and i'm not even referrring to the code layout. and then some people hate python mandaotry indent 😉

Reply:
Wait, Haskell is forcing you to duplicate the exact same code?!

Reply:
Yes, if you do the same thing in a different type context, you cannot reuse code, since top-level code is assigned a fixed type afaik.

Reply:
OK, so apparently it does work.

Reply:
\o/